src/anomalib/pipelines/benchmark/generator.py

```diff
@@ -0,0 +1,86 @@
+"""Benchmark job generator."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from argparse import SUPPRESS
+from collections.abc import Generator
+
+from jsonargparse import ArgumentParser, Namespace
+from jsonargparse._optionals import get_doc_short_description
+
+from anomalib.data import AnomalibDataModule, get_datamodule
+from anomalib.models import AnomalyModule, get_model
+from anomalib.pipelines.components import JobGenerator
+from anomalib.pipelines.components.actions import GridSearchAction, get_iterator_from_grid_dict
+from anomalib.utils.config import dict_from_namespace
+from anomalib.utils.logging import hide_output
+
+from .job import BenchmarkJob
+
+
+class BenchmarkJobGenerator(JobGenerator):
+    """Generate BenchmarkJob."""
+
+    def __init__(self, accelerator: str) -> None:
+        self.accelerator = accelerator
+
+    @property
+    def job_class(self) -> type:
+        """Return the job class."""
+        return BenchmarkJob
+
+    @staticmethod
+    def add_arguments(parser: ArgumentParser) -> None:
+        """Add job specific arguments to the parser."""
+        group = parser.add_argument_group("Benchmark job specific arguments.")
+        group.add_argument(
+            f"--{BenchmarkJob.name}.seed",
+            type=int | dict[str, list[int]],
+            default=42,
+            help="Seed for reproducibility.",
+        )
+        BenchmarkJobGenerator._add_subclass_arguments(group, AnomalyModule, f"{BenchmarkJob.name}.model")
+        BenchmarkJobGenerator._add_subclass_arguments(group, AnomalibDataModule, f"{BenchmarkJob.name}.data")
```

Comment:
I would prefer the use of `self` here for readability
```suggestion
        self._add_subclass_arguments(group, AnomalyModule, f"{BenchmarkJob.name}.model")
        self._add_subclass_arguments(group, AnomalibDataModule, f"{BenchmarkJob.name}.data")
```